src/Ext/Techno/Body.cpp

```diff
@@ -358,35 +372,51 @@ void TechnoExt::EatPassengers(TechnoClass* pThis)
 
 					if (pPassenger)
 					{
-						if (auto const pPassengerType = pPassenger->GetTechnoType())
+						VocClass::PlayAt(pData->PassengerDeletion_ReportSound, pThis->GetCoords(), nullptr);
+
+						if (pData->PassengerDeletion_Anim.isset())
 						{
-							VocClass::PlayAt(pData->PassengerDeletion_ReportSound, pThis->GetCoords(), nullptr);
+							const auto pAnimType = pData->PassengerDeletion_Anim.Get();
+							if (auto const pAnim = GameCreate<AnimClass>(pAnimType, pThis->Location))
+							{
+								pAnim->SetOwnerObject(pThis);
+								pAnim->Owner = pThis->Owner;
+							}
+						}
 
-							if (pData->PassengerDeletion_Anim.isset())
+						// Check if there is money refund
+						if (pData->PassengerDeletion_Soylent)
+						{
+							int nMoneyToGive = 0;
+
+							// Refund money to the Attacker
+							nMoneyToGive = pPassenger->GetTechnoType()->GetRefund(pPassenger->Owner, true);
```

Comment:
This can be merged into 1 line.

```suggestion
							// Refund money to the Attacker
							int nMoneyToGive = pPassenger->GetTechnoType()->GetRefund(pPassenger->Owner, true);
```